src/main/java/org/sagebionetworks/web/client/presenter/RejectReasonWidget.java

```diff
@@ -0,0 +1,151 @@
+    CallbackP<String> callback;
+
+    @Inject
+    public RejectReasonWidget (UserProfileAsyncHandler handler, RejectReasonView view) {
+        this.handler = handler;
+        this.view = view;
+        this.view.setPresenter(this);
+        this.userName = "";
+        this.callback = null;
+    }
+
+    private void setUserProfileName(UserProfile user) {
+        this.userName = DisplayUtils.getDisplayName(user);
+    }
+
+    /*
+        For testing purposes only relay the userName to assert
+        that its being properly found.
+     */
+    public String getUserName() {
+        return this.userName;
+    }
+
+    // TODO: Remove when feature is in prod, used for demo purposes in ComingSoon
+    public void setUserProfileName(String user) {
+        this.userName = user;
+    }
+
+    public void show(String userID, CallbackP<String> callback) {
+        handler.getUserProfile(userID, new AsyncCallback<UserProfile>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                view.showError("Could not find user id -- " + userID + "\nError -- " +  throwable.getMessage());
+                view.show();
+            }
+
+            @Override
+            public void onSuccess(UserProfile user) {
+                setUserProfileName(user);
+                view.show();
+            }
+        });
+        this.callback = callback;
+    }
+
+    // TODO: Remove when feature is in prod, used for demo purposes in ComingSoon
+    public void show(String userID) {
+        handler.getUserProfile(userID, new AsyncCallback<UserProfile>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                setUserProfileName("John Doe");
+                view.show();
+            }
+
+            @Override
+            public void onSuccess(UserProfile user) {
+                setUserProfileName(user);
+                view.show();
+            }
+        });
+    }
+
+    @Override
+    public void updateResponse() {
+        String output = "";
+
+        if (view.optionOneIsUsed()) {
+            output += "\n\t" + REJECT_TAKE_SYNAPSE_QZ + "\n";
+        }
+        if (view.optionTwoIsUsed()) {
+            output += "\n\t" + REJECT_ADD_INFO + "\n";
+        }
+        if (view.optionThreeIsUsed()) {
+            output += "\n\t" + REJECT_INITIAL_BOX + "\n";
+        }
+        if (view.optionFourIsUsed()) {
+            output += "\n\t" + REJECT_SUBMIT_DOCS + "\n";
+        }
+        if (view.optionFiveIsUsed()) {
+            output += "\n\t" + view.getCustomTextResponse() + "\n";
+        }
+
+        if (output.equals("") && view.getValue().equals("")) {
+            view.showError(ERROR_MESSAGE);
+            return;
+        } else {
+            view.clear();
+        }
+
+        view.setValue(TEMPLATE_HEADER_HELLO + this.getUserName() + TEMPLATE_HEADER_THANKS + output + "\n" + TEMPLATE_HEADER_SIGNATURE);
+    }
+
+    public String getValue () {
```

Comment:
Can remove.  The Value (rejection reason) is sent back to the parent widget in onSave (via the CallbackP() that was given in show()).  

There are many other ways to send a message back, including going through the event bus (but that's probably over-engineering this widget).